### PR TITLE
Shifting labels to enable learning next token prediction

### PR DESCRIPTION
**Pull request by LiorZ**
Dear lucidrains,
Many thanks for your amazing contribution to the community.
In the following pull request I modified the code so that the labels to the right during training before calculating the loss to enable next-token prediction (otherwise, the loss is calculated for the token, which leads to learning the trivial, current token)

L

### CODE REVIEW

**Comment by lucidrains** [comment]
@LiorZ hey Lior

it should be shifted already https://github.com/lucidrains/perceiver-ar-pytorch/blob/main/perceiver_ar_pytorch/autoregressive_wrapper.py#L83
